### PR TITLE
HHH-15846 Option to add @SuppressFBWarnings annotation on generated code

### PR DESCRIPTION
**Pull request by NicklasWallgren**
Lombok has an option to add `@SuppressFBWarnings` to generated code via `lombok.extern.findbugs.addSuppressFBWarnings`.

It would be really helpful if the hibernate gradle/maven enhance plugins would provide a similar option.

We're currently struggling with warnings related to hibernate generated code in our entities, in SpotBugs.

```
NP | Non-null field $_hibernate_attributeInterceptor is not initialized by new <entity>
NP | Non-null field $_hibernate_entityEntryHolder is not initialized by new <entity>
NP | Non-null field $_hibernate_nextManagedEntity is not initialized by new <entity>
NP | Non-null field $_hibernate_previousManagedEntity is not initialized by new <entity>
NP | Non-null field $_hibernate_tracker is not initialized by new <entity>
```

Discussion on GitHub
Enhance plugin - Option to add @SuppressFBWarnings annotation · Discussion #5668 · hibernate/hibernate-orm

### CODE REVIEW

File: hibernate-core/hibernate-core.gradle [hunk 1]
```diff
@@ -61,6 +61,8 @@ dependencies {
         transitive = true
     }
     testImplementation "joda-time:joda-time:2.3"
+    testImplementation testLibs.archunitJUnit5
```
Comment (NicklasWallgren) [inline hibernate-core/hibernate-core.gradle:64]: I'm using ArchUnit to be able to assert annotations with retention policy `RetentionPolicy.CLASS`.

File: hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/CodeTemplates.java [hunk 2]
```diff
@@ -543,6 +543,23 @@ static Collection<Object> getter(Object self) {
...
 
+	/**
+	 * Used to suppress <a href="https://spotbugs.readthedocs.io">SpotBugs</a> warnings.
+	 */
+	@Retention(RetentionPolicy.CLASS)
+	@interface SuppressFBWarnings {
+		/**
+		 * @see "https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html"
+		 */
+		String[] value();
```
Comment (NicklasWallgren) [inline hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/CodeTemplates.java:554]: We might be able to remove `value()`. It works fine without it.

**Comment by NicklasWallgren** [comment]
@gavinking Are you able to review this PR, or any of the other contributors? 🙂
Thanks

**Comment by gavinking** [comment]
> @gavinking Are you able to review this PR, or any of the other contributors? 🙂
> Thanks

Hi, I don't know what `@SuppressFBWarnings` or why you want this, and neither the JIRA issue nor the discussion explain it...

**Comment by NicklasWallgren** [comment]
> > @gavinking Are you able to review this PR, or any of the other contributors? 🙂
> > Thanks
> 
> Hi, I don't know what `@SuppressFBWarnings` is, nor why you want this, and neither the JIRA issue nor the discussion explain it...

Ah sorry, I mixed you up with @Sanne.

**Comment by sebersole** [comment]
@NicklasWallgren Can you rebase this on main?  And I will take a look.

**Comment by sebersole** [comment]
Also, the [discussion](https://github.com/hibernate/hibernate-orm/discussions/5668#discussioncomment-4745386) mentioned `@Generated` but I do not see that handled here.  Is that a different PR or just not added?

**Comment by sebersole** [comment]
@NicklasWallgren Could you explain how an annotation you are adding into Hibernate here in this PR somehow affects those other libraries?  How do FindBugs, etc know to look for this annotation?  

Could we not just leverage `java.lang.SuppressWarnings` for this?

**Comment by gavinking** [comment]
@NicklasWallgren has not responded to Steve's questions, and the branch now has conflicts, so perhaps we can just close this?

**Comment by beikov** [comment]
Please create a new PR once you're willing to fixup the issues.

**Pull request closed without merging**